ODLA/platforms/odla_popart/odla_compute.cc

```diff
@@ -147,28 +167,35 @@ odla_status odla_CreateContext(odla_context* context) {
 
   // Create InferenceSession
   auto proto = g_comp->builder->getModelProto();
-  auto session = popart::InferenceSession::createFromOnnxModel(
+  g_comp->session = popart::InferenceSession::createFromOnnxModel(
       proto, data_flow, device, popart::InputShapeInfo(), *opts);
-  *context = new _odla_context(g_comp, std::move(session));
 
   // Compile graph, create engine and load into the IPU
   // use compileAndExport() to frozen engine to specified path
-  (*context)->session->prepareDevice();
+  g_comp->session->prepareDevice();
   // Init seed
-  (*context)->session->setRandomSeed(0);
+  g_comp->session->setRandomSeed(0);
   // Copy weights from host to IPU
-  (*context)->session->weightsFromHost();
+  g_comp->session->weightsFromHost();
 
   return ODLA_SUCCESS;
 }
 
 odla_status odla_DestroyContext(odla_context ctx) {
-  delete (ctx);
+  if (ctx != nullptr) {
+      delete (ctx);
+  }
   return ODLA_SUCCESS;
 }
 
 odla_status odla_DestroyComputation(odla_computation comp) {
-  // g_comp.reset();
+  if (comp->session != nullptr) {
+    comp->session->getDevice().getDeviceInfo()->detach();
+    comp->session.reset();
+  }
+  if (g_comp = nullptr) {
```

Comment:
something like
```
if (g_comp == comp) { g_comp == nullptr;}
delete comp;
```
?

Reply:
It seems that set `g_comp=nullptr` and `delete comp` or `comp = nullptr` together will cause a segment error.